core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java

```diff
@@ -77,8 +77,32 @@ class FormatElementEnumTest {
     assertFormatElement(FormatElementEnum.FF6, "2014-09-30T10:00:00.123456Z", "000123");
   }
 
-  @Test void testIW() {
+  @Test void testID() {
+    assertFormatElement(FormatElementEnum.ID, "2014-09-30T10:00:00Z", "2");
+  }
+
+  @Test
+  void testIW() {
     assertFormatElement(FormatElementEnum.IW, "2014-09-30T10:00:00Z", "40");
+    // edge case where ISO WEEK != WEEK
```

Comment:
I have discovered that the Java Calendar class doesn't handle correctly some operations with dates before the Gregorian calendar introduction. Can you please add some tests for dates in that range?
https://issues.apache.org/jira/projects/CALCITE/issues/CALCITE-6252
Can you also confirm that the values have been validated against BigQuery?

Reply:
I added additional test cases, all manually validated against BigQuery.